### PR TITLE
Rework CUDA compiler checks

### PR DESCRIPTION
**Pull request by davebayer**
This PR introduces a new CUDA compiler handling, following the C++ compiler checks.

Changes:

* `_CCCL_CUDA_COMPILER_{NAME}` macros now expands to the CUDA compiler version, same as the c++ compilers, e. g. for NVHPC compiler: `_CCCL_COMPILER_NVHPC == _CCCL_CUDA_COMPILER_NVHPC`
* to check CUDA compiler (and version), use `_CCCL_CUDA_COMPILER(NAME [, OP, MAJ [, MIN]])` macro, same as way as for the C++ compilers
* `_CCCL_CUDACC` expands to the CUDA Toolkit version (that's what `_CCCL_CUDACC_VER` did in the past)
* `_CCCL_CUDACC_{BELOW,AT_LEAST}` now allows omiting the minor version
* the `_CCCL_CUDA_COMPILER` macro is now always defined and the old behaviour is replaced by the `_CCCL_HAS_CUDA_COMPILER` macro to avoid naming issues

### CODE REVIEW

File: cudax/include/cuda/experimental/__async/continue_on.cuh [hunk 1]
```diff
@@ -38,10 +38,10 @@ namespace cuda::experimental::__async
 {
 struct continue_on_t
 {
-#if !defined(_CCCL_CUDA_COMPILER_NVCC)
+#if !_CCCL_CUDA_COMPILER(NVCC)
 
 private:
-#endif // _CCCL_CUDA_COMPILER_NVCC
+#endif // _CCCL_CUDA_COMPILER(NVCC)
```
Comment (miscco) [inline cudax/include/cuda/experimental/__async/continue_on.cuh]:
```suggestion
#endif // !_CCCL_CUDA_COMPILER(NVCC)
```

**Review by miscco — APPROVED** [review]
I fixed some minor issues with comments

**Review by bernhardmgruber — APPROVED** [review]

**Comment by bernhardmgruber** [comment]
Thank you for this improvement!

**Pull request merged**